server/models/product.js

```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+class Product {
+  constructor(id, farmerID, name, description, category) {
+    this.id = id;
+    this.farmerID = farmerID;
+    this.name = name;
+    this.description = description;
+    if (Object.values(ProductCategory.Category).includes(category))
+      this.category = category;
+  }
+
+  static fromMongoJSON(json) {
+    return new Product(
+      json._id,
+      json.farmerID,
+      json.name,
+      json.description,
+      json.category
+    );
+  }
+}
+
+class ProductCategory {
```

Comment:
You can put the ProductCategory object outside of a class.

```suggestion
const ProductCategory = {
     FRUIT: "fruit",
     VEGETABLES: "vegetables",
     SPREADABLECREAMS: "spreadable creams",
     MEAT: "meat",
     EGGS: "eggs",
     MILK: "milk",
};
```

And then export it.